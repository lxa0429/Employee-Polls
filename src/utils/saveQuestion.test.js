import { _saveQuestion } from './_DATA'

describe('_saveQuestion', () => {
  it('should return the saved question and match thebsnapshot with all expected fields populated when correctly formatted data is passed', async () => {
    const question = {
      optionOneText: 'Option One',
      optionTwoText: 'Option Two',
      author: 'sarahedo'
    };

    const savedQuestion = await _saveQuestion(question);

    expect(savedQuestion).toMatchSnapshot();

    expect(savedQuestion).toHaveProperty('id');
    expect(savedQuestion).toHaveProperty('timestamp');
    expect(savedQuestion).toHaveProperty('author', question.author);
    expect(savedQuestion).toHaveProperty('optionOne');
    expect(savedQuestion.optionOne).toHaveProperty('text', question.optionOneText);
    expect(savedQuestion.optionOne).toHaveProperty('votes', []);
    expect(savedQuestion).toHaveProperty('optionTwo');
    expect(savedQuestion.optionTwo).toHaveProperty('text', question.optionTwoText);
    expect(savedQuestion.optionTwo).toHaveProperty('votes', []);
  });
});

describe('_saveQuestion', () => {
    it('should return an error if incorrect data is passed to the function', async () => {
      const question = {
        optionOneText: 'Option One'
        // Missing optionTwoText and author
      };
  
      await expect(_saveQuestion(question)).rejects.toEqual(
        "Please provide optionOneText, optionTwoText, and author"
      );
    });
  });