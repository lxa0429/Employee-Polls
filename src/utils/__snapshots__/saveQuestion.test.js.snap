// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_saveQuestion should return the saved question and match thebsnapshot with all expected fields populated when correctly formatted data is passed 1`] = `
Object {
  "author": "sarahedo",
  "id": "4tl36ldlbyj9lr54usiw6",
  "optionOne": Object {
    "text": "Option One",
    "votes": Array [],
  },
  "optionTwo": Object {
    "text": "Option Two",
    "votes": Array [],
  },
  "timestamp": 1723990455549,
}
`;
